Source/WebKit/Shared/RTCNetwork.serialization.in

```diff
@@ -0,0 +1,61 @@
+# Copyright (C) 2023 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#if USE(LIBWEBRTC)
+
+[CustomHeader] struct WebKit::RTC::Network::IPAddress {
+    std::variant<WebKit::RTC::Network::IPAddress::UnspecifiedFamily, uint32_t, std::array<uint32_t, 4>> value;
+};
+
+[Nested] struct WebKit::RTC::Network::IPAddress::UnspecifiedFamily {
+};
+
+[CustomHeader] struct WebKit::RTC::Network::InterfaceAddress {
+    WebKit::RTC::Network::IPAddress address;
+    int ipv6Flags;
+};
+
+[CustomHeader] struct WebKit::RTC::Network::SocketAddress {
+    Vector<char> hostname;
+    WebKit::RTC::Network::IPAddress ip;
+    uint16_t port;
+    int scopeID;
+    bool literal;
```

Comment:
Why not reuse IsUnresolvedIP here for SocketAddress?
It should allow to map precisely to what the current code does and might not require changing libwebrtc, which is always a bit of a burden.

We could reuse the code from SocketAddress::decode in SocketAddress::rtcAddress()

Reply:
That caused a test timeout last time for an unknown reason.